contracts/examples/WitnetRandomness.sol

```diff
@@ -244,9 +244,9 @@ contract WitnetRandomness
         RandomizeData storage _data = __randomize_[_block];
         if (_data.witnetQueryId != 0) {
             _usedFunds = _witnetUpgradeReward(_data.witnetQueryId);
-            if (_usedFunds > 0) {
-                payable(msg.sender).transfer(msg.value - _usedFunds);
-            }
+        }
+        if (_usedFunds > 0) {
+            payable(msg.sender).transfer(msg.value - _usedFunds);
```

Comment:
Btw, I struggle to follow this logic completely. In the case in which `_usedFunds` is `0`, wouldn't this contract drain the whole `msg.value`? Shouldn't the `transfer` be conditional on `msg.value > _usedFunds` instead of `_usedFunds > 0`?

Reply:
This transfer is the refund to the user (msg.sender), the user already sent msg.value to the contract, so this transfer returns the unused funds back to the user.

Reply:
True, should be `if _usedFunds < msg.value` instead?

Reply:
Yes, I think so, or `msg.value > _usedFunds` as I wrote before

Reply:
> I can't see how this change make any difference in practice, given that `_usedFunds > 0` can only be true if the first conditional statement is also true.

Right, bad fix indeed. Taking this piece out from where it was, requires changing condition to: `if (_usedFunds < msg.Value) { ... `

Reply:
Great, now the separate conditional does make sense to me. Resolved!